prometheus/metrics/types.go

```diff
@@ -0,0 +1,144 @@
+package metrics
+
+import (
+	kitprometheus "github.com/go-kit/kit/metrics/prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/go-kit/kit/metrics/discard"
+	"github.com/go-kit/kit/metrics"
+)
+
+type Counter metrics.Counter
+type Gauge metrics.Gauge
+
+type CounterSet struct {
+	itLvl []string
+	c     Counter
+}
+
+type GaugeSet struct {
+	itLvl []string
+	g     Gauge
+}
+
+func (cs *CounterSet) Add(delta float64, labelValues ...string) {
```

Comment:
I will leave u a comment on this line of code but it applies to the whole metrics pkg. I think u made it bit too abstract and the other way around. Exposing internals and then incrementing them.

Could u replace this implementation as we discussed please with this style?:

```
func (m Metrics) DeliverTxsTotal(code int) {
	m.checkErrTxsTotal.Set()
	m.checkErrTxsTotal.Add()
}
```

It solves the following 3 issues:

- clear method definition, u know what metric values to pass
- no need to define and maintain labels keys in 2 places
- no need for metrics provider

Reply:
Done, there is an entire refactor around it